src/rules/effects/avoid-cyclic-effects.ts

```diff
@@ -104,7 +103,7 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
         return memberActionTypes
       }
       const symbol = typeChecker.getPropertyOfType(type, 'type')
-      if (!symbol) {
+      if (!symbol?.valueDeclaration) {
```

Comment:
I suppose this was the fix for the "todo test"?
If so, I suppose we should commit this as a fix, so it releases a new version.

Reply:
Not really. This was changed because now [`Symbol.valueDeclaration`](https://github.com/microsoft/TypeScript/blob/e00722f262f8a84c27f01e2b6a41eecb87c4afd4/src/compiler/types.ts#L4833) is possibly `undefined` and `TypeChecker.getTypeOfSymbolAtLocation` only accepts a `Node`, not `Node | undefined`.

Note that it was reported in https://github.com/microsoft/TypeScript/issues/24706 and fixed in https://github.com/microsoft/TypeScript/pull/43033). Also, this was also fixed in the [no-cyclic-action.ts](https://github.com/cartant/eslint-plugin-rxjs/blob/main/source/rules/no-cyclic-action.ts) in this [commit](https://github.com/cartant/eslint-plugin-rxjs/commit/fee48bceda7d214d3ef52234428620d302e5ee7f)

---

That said, I must confess that I didn't do anything to fix that test case. I was simply testing all the tests to see if everything was working as before and I realized that one case was failing, so I came across to that comment and I simply moved it to invalid section 😆.

Reply:
😅 that's why I couldn't figure out what the actual fix was.
Thanks for the answer!